vendors/espressif/esp-idf/components/esp32/esp_adapter.c

```diff
@@ -523,24 +522,24 @@ wifi_osi_funcs_t g_wifi_osi_funcs = {
     ._mutex_lock = mutex_lock_wrapper,
     ._mutex_unlock = mutex_unlock_wrapper,
     ._queue_create = queue_create_wrapper,
-    ._queue_delete = vQueueDelete,
+    ._queue_delete = (void(*)(void *))vQueueDelete,
     ._queue_send = queue_send_wrapper,
     ._queue_send_from_isr = queue_send_from_isr_wrapper,
     ._queue_send_to_back = queue_send_to_back_wrapper,
     ._queue_send_to_front = queue_send_to_front_wrapper,
     ._queue_recv = queue_recv_wrapper,
-    ._queue_msg_waiting = uxQueueMessagesWaiting,
-    ._event_group_create = xEventGroupCreate,
-    ._event_group_delete = vEventGroupDelete,
-    ._event_group_set_bits = xEventGroupSetBits,
-    ._event_group_clear_bits = xEventGroupClearBits,
+    ._queue_msg_waiting = (uint32_t(*)(void *))uxQueueMessagesWaiting,
```

Comment:
Again I would really like to avoid casting function pointers at all cost, why do we get a warning here?

Reply:
Similar is the case with `._queue_msg_waiting`. Parameter to `_queue_msg_waiting` is `void *`, while the parameter to the function `uxQueueMessagesWaiting` is `const QueueHandle_t`.